### PR TITLE
Add explicit netstandard2.0 support

### PR DESCRIPTION
**Pull request by am11**
Although netstandard1.x libs are compatible with higher version of
standards, it is a good idea to target each standard specifically
if the dependencies are different. In case of NETSTANDARD 2.0,
we don't need additional dependencies other than the default
`NETStandard.Library` which is a
[flat package](https://stackoverflow.com/a/43999545/863980).

Also removed the mono-specific solution and project files, as the
latest Mono uses CSC (C# Roslyn compiler from dotnet/roslyn repo),
instead of MCS and MSBuild (from microsoft/MSBuild repo) instead of
xbuild. See http://www.mono-project.com/docs/about-mono/releases/5.0.0
for details.

### CODE REVIEW

**Comment by am11** [comment]
TraivsCI seems to be disabled atm for this repo: https://travis-ci.org/maxmind/MaxMind-DB-Reader-dotnet.

File: appveyor.yml [hunk 1]
```diff
@@ -17,13 +17,17 @@ environment:
       TEST_FRAMEWORK: net452
       LIB_FRAMEWORK: net45
     - CONFIGURATION: Debug
-      CONSOLE_FRAMEWORK: netcoreapp1.0
-      LIB_FRAMEWORK: netstandard1.4
-      TEST_FRAMEWORK: netcoreapp1.0
```
Comment (am11) [inline appveyor.yml]: I was thinking if it makes sense to only configure the build+release jobs for the latest framework and release for older ones.

**Comment by am11** [comment]
TravisCI seems to still have some issue here https://travis-ci.org/maxmind/MaxMind-DB-Reader-dotnet/branches, but working against my fork https://travis-ci.org/am11/MaxMind-DB-Reader-dotnet.

Maybe in repo settings there, the PR trigger is off:

image

**Comment by oschwald** [comment]
Weird. "Build pull request updates" was enabled. I deleted the integration from GitHub and then re-enabled it again in Travis. Hopefully it should work now. Could you try pushing something?

**Comment by am11** [comment]
Yay, it worked! 🌮

**Comment by am11** [comment]
@oschwald, with dotnet cli 2.0, if we have full mono or only reference assemblies of some desktop TFM (say net45, net32, net20) installed on Unix, then we can use dotnet `restore`, `build`, `test` commands for those TFMs on Unix as well. Tried it here: https://github.com/neris/NGettext/blob/a701ac6/.travis.yml. But in our matrix, we are not installing dotnet-cli in mono jobs that's why we cant use dotnet commands which makes sense as well.

**Comment by oschwald** [comment]
Thanks!

**Pull request merged**